### ISSUE TITLE
Some TSLint errors aren't shown in VSCode

### ISSUE THREAD

**Issue by JamesMessinger** [Microsoft/vscode-tslint#408]
- **tslint Version:** 5.11.0
- **typescript Version:** 3.1.6

# Steps to Reproduce:
Here is a sample repo with a minimalistic reproduction of the bug: https://github.com/JamesMessinger/tslint-vscode-bug-demo

In the above repo, there are only two TSLint rules enabled:

```yaml
rules:
  no-empty: true
  no-void-expression: true
```

And a single TypeScript code file that contains violations of each of these rules:

```typescript
export function add(a: number, b: number): number {
  console.log(empty());                             // <--- violates the "no-void-expression" rule
  return a + b;
}

export function empty(): void { }                   // <--- violates the "no-empty" rule
```

When running TSLint from the command-line, it shows both errors, as expected:

![cli-screenshot](https://user-images.githubusercontent.com/3453903/48557575-c613b780-e8ac-11e8-9a3f-b44a907b4c5d.png)

But in VSCode, only the "no-empty" error on line 6 is shown.  Notice that it _does not_ show the "no-void-expression" error on line 2:

![vscode-screenshot](https://user-images.githubusercontent.com/3453903/48557591-ca3fd500-e8ac-11e8-84ef-a5f25a368efc.png)

**Comment by JamesMessinger** [comment]
I've found the same problem with many other TSLint rules as well, where they're not shown in VSCode.  But I'm keeping things simple here by just demonstrating one rule that's not working (`no-void-expression`)

**Comment by egamma** [comment]
VS Code tslint does not support rules that require type information [pls see](https://github.com/Microsoft/vscode-tslint/blob/master/tslint/README.md#how-can-i-use-tslint-rules-that-require-type-information). You get a warning for this in the output channel (that can easily be overlooked).

To overcome this limitation we are currently working on a new implementation that runs tslint as a TS Server plugin. Please give this extension a try: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin.

**Comment by JamesMessinger** [comment]
Thanks @egamma!  I'll try the new extension out
